src/builder-context.c

```diff
@@ -1193,6 +1193,58 @@ builder_context_get_sdk_config (BuilderContext *self)
   return self->sdk_config;
 }
 
+gboolean
+builder_context_create_state_dir (BuilderContext *self,
+                                  GError        **error)
+{
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+  gboolean exists = g_file_query_exists (self->state_dir, NULL);
+
+  if (!exists && !g_file_make_directory_with_parents (self->state_dir, NULL, error))
+  {
+    flatpak_fail (error, "Could not create state dir");
+    return FALSE;
+  }
```

Comment:
```suggestion
  if (!g_file_make_directory_with_parents (self->state_dir, NULL, error))
  {
    if (g_error_matches(error, G_IO_ERROR, G_IO_ERROR_EXISTS))
      return TRUE;

    flatpak_fail (error, "Could not create state dir");
    return FALSE;
  }
```

Race condition between check and creation.